File: tests/fixtures/typescript/authService.ts
import { JwtHelper } from './jwtHelper';
import type { Token } from '../types/token';
import * as crypto from 'node:crypto';
import defaultConfig from './config';

export interface IAuthService {
  refresh(token: string): Promise<Token>;
}

export type RefreshResult = { token: string; expiresAt: number };

export class AuthService implements IAuthService {
  private readonly helper: JwtHelper;

  constructor(helper: JwtHelper) {
    this.helper = helper;
  }

  async refresh(token: string): Promise<Token> {
    return this.helper.rotate(token);
  }

  get issuer(): string {
    return defaultConfig.issuer;
  }
}

export function rotateRefreshToken(token: string): string {
  return crypto.createHash('sha256').update(token).digest('hex');
}

export const buildCacheKey = (userId: string): string => `session:${userId}`;

File: tests/fixtures/python/auth_service.py
import hashlib
from dataclasses import dataclass

from .jwt_helper import JwtHelper
from ..types.token import Token
from api.config import settings


@dataclass
class RefreshResult:
    token: str
    expires_at: int


class AuthService:
    def __init__(self, helper: JwtHelper) -> None:
        self._helper = helper

    async def refresh(self, token: str) -> Token:
        return await self._helper.rotate(token)

    @property
    def issuer(self) -> str:
        return settings.issuer


def rotate_refresh_token(token: str) -> str:
    return hashlib.sha256(token.encode()).hexdigest()
